inst/stan/lm-claret-bruno/link_dsld.stan

```diff
@@ -0,0 +1,42 @@
+
+functions {
+    //
+    // Source - lm-claret-bruno/link_dsld.stan
+    //
+    matrix link_dsld_contrib(
+        matrix time,
+        matrix link_function_inputs
+    ) {
+        int nrows = rows(link_function_inputs);
+        int ncols = cols(time);
+        vector[nrows] ind_b = link_function_inputs[,1];
+        vector[nrows] ind_g = link_function_inputs[,2];
+        vector[nrows] ind_c = link_function_inputs[,3];
+        vector[nrows] ind_p = link_function_inputs[,4];
+        
+        // Here we assume that psi's are replicated along the rows of the time matrix.
+        matrix[nrows, ncols] ind_b_matrix = rep_matrix(ind_b, ncols);
+        matrix[nrows, ncols] ind_g_matrix = rep_matrix(ind_g, ncols);
+        matrix[nrows, ncols] ind_c_matrix = rep_matrix(ind_c, ncols);
+        matrix[nrows, ncols] ind_p_matrix = rep_matrix(ind_p, ncols);
+
+        matrix[nrows, ncols] result_lt0 = fmin(
```

Comment:
unrelated idea - it would be better to centralize these `fmin(8000.0, .)` calls in something like `safecap()` such that the hardcoded 8000 value could be replaced in one go, and in order to enhance readability